R/CDISCTealData.R

```diff
@@ -1,270 +0,0 @@
-#
-# test_dataset <- cdisc_dataset("ADAE", synthetic_cdisc_data("latest")$adae)
-# test_data <- cdisc_data(test_dataset, test_adsl)
-# get_keys(test_data, "ADAE") # returns [1] "STUDYID" "USUBJID" "ASTDTM"  "AETERM"  "AESEQ"
-#' ```
-#' @inheritParams teal_data
-#' @param ... (`TealDataConnector`, `TealDataset` or
-#'   `TealDatasetConnector`) elements to include where `ADSL` data is mandatory.
-#' @param join_keys (`JoinKeys`) or a single (`JoinKeySet`)\cr
-#'   (optional) object with datasets column names used for joining.
-#'   If empty then it would be automatically derived basing on intersection of datasets primary keys
-#'
-#' @return a `CDISCTealData` object
-#'
-#' @details This function checks if there were keys added to all data sets
-#'
-#' @export
-#'
-#' @examples
-#' library(scda)
-#'
-#' latest_data <- synthetic_cdisc_data("latest")
-#' ADSL <- latest_data$adsl
-#' ADTTE <- latest_data$adtte
-#'
-#' # basic example
-#' cdisc_data(
-#'   cdisc_dataset("ADSL", ADSL),
-#'   cdisc_dataset("ADTTE", ADTTE),
-#'   code = 'ADSL <- synthetic_cdisc_data("latest")$adsl
-#'           ADTTE <- synthetic_cdisc_data("latest")$adtte',
-#'   check = TRUE
-#' )
-#'
-#' # Example with keys
-#' cdisc_data(
-#'   cdisc_dataset("ADSL", ADSL,
-#'     keys = c("STUDYID", "USUBJID")
-#'   ),
-#'   cdisc_dataset("ADTTE", ADTTE,
-#'     keys = c("STUDYID", "USUBJID", "PARAMCD"),
-#'     parent = "ADSL"
-#'   ),
-#'   join_keys = join_keys(
-#'     join_key(
-#'       "ADSL",
-#'       "ADTTE",
-#'       c("STUDYID" = "STUDYID", "USUBJID" = "USUBJID")
-#'     )
-#'   ),
-#'   code = 'ADSL <- synthetic_cdisc_data("latest")$adsl
-#'           ADTTE <- synthetic_cdisc_data("latest")$adtte',
-#'   check = TRUE
-#' )
-cdisc_data <- function(...,
-                       join_keys,
-                       code = "",
-                       check = FALSE) {
-  x <- CDISCTealData$new(..., check = check, join_keys = join_keys)
-  if (length(code) > 0 && !identical(code, "")) {
-    x$set_pull_code(code = code)
-  }
-
-  x$check_reproducibility()
-  x$check_metadata()
-
-  return(x)
-}
-
-#' Load `CDISCTealData` object from a file
-#'
-#' @description `r lifecycle::badge("experimental")`
-#'
-#' @inheritParams teal_data_file
-#'
-#' @return `CDISCTealData` object
-#'
-#' @export
-#'
-#' @examples
-#' file_example <- tempfile(fileext = ".R")
-#' writeLines(
-#'   text = c(
-#'     "library(scda)
-#'
-#'      # code>
-#'      ADSL <- synthetic_cdisc_data('latest')$adsl
-#'      ADTTE <- synthetic_cdisc_data('latest')$adtte
-#'
-#'      cdisc_data(
-#'           cdisc_dataset(\"ADSL\", ADSL), cdisc_dataset(\"ADTTE\", ADTTE),
-#'           code = \"ADSL <- synthetic_cdisc_data('latest')$adsl
-#'                   ADTTE <- synthetic_cdisc_data('latest')$adtte\",
-#'           check = FALSE
-#'      )
-#'      # <code"
-#'   ),
-#'   con = file_example
-#' )
-#'
-#' cdisc_data_file(file_example)
-cdisc_data_file <- function(path, code = get_code(path)) {
```

Comment:
should this be deprecated? happy to remove though

Reply:
fair enough, deprecated with `deprecate_warn`.